src/www/scripts/app.js

```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
 import FileStore from "./file-store";
 import LocalStore from "./local-store";
 
+function Emitter() {
+    let events = {};
+    return {
+        on: (eventName, callback) => {
+            events[eventName] = [...(events[eventName] || []), callback];
+            return this;
+        },
+        emit: async (eventName, ...args) => {
+            const callbacks = events[eventName];
+            if (callbacks && callbacks.length > 0) {
+                return await Promise.all(callbacks.map(callback => callback(...args)));
```

Comment:
2. Then you can just `return Promise.all` here

Reply:
Done